src/selector.js

```diff
@@ -436,6 +437,15 @@ function setDocument( node ) {
 	documentElement = document.documentElement;
 	documentIsHTML = !jQuery.isXMLDoc( document );
 
+	// Support: IE 11+
+	// IE 11 doesn't find elements on a `[name='']` query in some cases.
+	// Adding a temporary attribute to the document before the selection works
+	// around the issue.
+	if ( isIE && documentElement ) {
+		documentElement.setAttribute( "name", "" );
```

Comment:
I guess I should check if there's such an attribute already set and - if it is - restore it afterwards. `documentElement` with a `name` attribute would be quite a weird thing to do but it's not impossible.

Reply:
I wonder if the problem is already resolved if the `documentElement` already has a `name` attribute.

Reply:
@timmywil It looks like the problem doesn't exist if an _empty_ name attribute is already present on the document. However, a non-empty attribute doesn't help so this is all needed.